### ISSUE TITLE
[bug] Unable to use picojson installed to `/usr/include/picojson.h`

### ISSUE THREAD

**Issue by olifre** [Thalhammer/jwt-cpp#213]
**Describe the bug**
`picojson` defaults to installation into `<prefix>/include/picojson.h`. 
However, `jwt-cpp` tries to include from `<includepath>/picojson/picojson.h` which will not be used by a standard `picojson` installation. 

**Desktop (please complete the following information):**
 - OS: Gentoo Linux with `picojson` installed from [upstream](https://github.com/kazuho/picojson) via system package `dev-cpp/picojson`-
 - Compiler: `gcc (Gentoo 11.2.0 p1) 11.2.0`

**Comment by prince-chrismc** [comment]
You are absolutely correct, https://github.com/kazuho/picojson/blob/111c9be5188f7350c2eac9ddaedd8cca3d7bf394/Makefile#L21

Do you have any suggestions on how we can support both?

**Comment by olifre** [comment]
I believe the most flexible approach would be (but that needs some more changes):

* Make `JWT_EXTERNAL_PICOJSON` a bit more flexible, i.e. in case it is set, allow the user to specify a `PICOJSON_INCLUDE_PATH` from outside instead of `find_package`. 

* Embed that path (which could default to `picojson/` so it works with the bundled version) into the `include` statements. That could be done via a preprocessor macro and a header config file created by `CMake` for example.

**Comment by olifre** [comment]
> jwt-cpp could provide a custom CMake module that searches both locations and setups the target correctly?

That sounds like a good solution :-). 

My experience with hunter, conan or vcpkg is sadly zero, I have first heard of these in this project. As a quick hack, I have for now used the following approach in Gentoo packaging:
https://github.com/gentoo/guru/blob/47471944d857275e82983460b6cd720257510df7/dev-cpp/jwt-cpp/jwt-cpp-0.6.0.ebuild#L31-L38

I'm not sure I'll have sufficient time for a PR in the near future, but I believe the approach is a good one :+1: .

**Comment by prince-chrismc** [comment]
Note for whoever gets to this first, #195 is closely related and we should write some docs to help the community
